### PR TITLE
*: add source-id checker

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 
ref https://github.com/pingcap/dm/issues/141

### What is changed and how it works?
add source-id length check in
- dm-worker subtask adjust
- dm-worker config verify
- dm-ansible check_config

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
    - modify a source-id more than 32 chars in dm-ansible `inventory.ini`
    - run `ansible-playbook rolling_update.yml` or `ansible-playbook deploy.yml` 

Related changes
 - Need to update the documentation

### CODE REVIEW

**Comment by amyangfei** [comment]
PTAL @GregoryIan @csuzhangxc

File: dm/config/checking_item.go [hunk 1]
```diff
@@ -49,6 +49,9 @@ var AllCheckingItems = map[string]string{
 	ShardAutoIncrementIDChecking: "conflict auto increment ID of shard tables checking item",
 }
 
+// MaxSourceIDLength is the max length for dm-worker source id
+var MaxSourceIDLength = 32
```
Comment (csuzhangxc) [inline dm/config/checking_item.go]: change to `const`?

File: dm/worker/config_test.go [hunk 2]
```diff
@@ -69,3 +73,82 @@ func (t *testServer) TestConfig(c *C) {
...
+
+func (t *testServer) TestConfigVerify(c *C) {
+	newConfig := func() *Config {
+		cfg := NewConfig()
+		c.Assert(cfg.Parse([]string{"-config=./dm-worker.toml", "-relay-dir=./xx"}), IsNil)
+		return cfg
+	}
+	testCases := []struct {
+		genFunc     func() *Config
+		hasError    bool
```
Comment (csuzhangxc) [inline dm/worker/config_test.go]: use non-empty `errorFormat` indicate `hasError` and save this field?

**Pull request merged**